packages/authorization-process/test/api/addProducerKeychainEService.test.ts

```diff
@@ -0,0 +1,132 @@
+      .post(`/producerKeychains/${producerKeychainId}/eservices`)
+      .set("Authorization", `Bearer ${token}`)
+      .set("X-Correlation-Id", generateId())
+      .send({ eserviceId: mockEService.id });
+
+  const authorizedRoles: AuthRole[] = [authRole.ADMIN_ROLE];
+  it.each(authorizedRoles)(
+    "Should return 204 for user with role %s",
+    async (role) => {
+      const token = generateToken(role);
+      const res = await makeRequest(token, mockProducerKeychain.id);
+      expect(res.status).toBe(204);
+    }
+  );
+
+  it.each(
+    Object.values(authRole).filter((role) => !authorizedRoles.includes(role))
+  )("Should return 403 for user with role %s", async (role) => {
+    const token = generateToken(role);
+    const res = await makeRequest(token, mockProducerKeychain.id);
+    expect(res.status).toBe(403);
+  });
+
+  it("Should return 404 for producerKeychainNotFound", async () => {
+    authorizationService.addProducerKeychainEService = vi
+      .fn()
+      .mockRejectedValue(producerKeychainNotFound(mockProducerKeychain.id));
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token, mockProducerKeychain.id);
+    expect(res.status).toBe(404);
+  });
+
+  it("Should return 404 for eserviceNotFound", async () => {
+    authorizationService.addProducerKeychainEService = vi
+      .fn()
+      .mockRejectedValue(eserviceNotFound(mockEService.id));
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token, mockProducerKeychain.id);
+    expect(res.status).toBe(404);
+  });
+
+  it("Should return 409 for eserviceAlreadyLinkedToProducerKeychain", async () => {
+    authorizationService.addProducerKeychainEService = vi
+      .fn()
+      .mockRejectedValue(
+        eserviceAlreadyLinkedToProducerKeychain(
+          mockEService.id,
+          mockProducerKeychain.id
+        )
+      );
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token, mockProducerKeychain.id);
+    expect(res.status).toBe(409);
+  });
+
+  it("Should return 403 for organizationNotAllowedOnProducerKeychain", async () => {
+    authorizationService.addProducerKeychainEService = vi
+      .fn()
+      .mockRejectedValue(
+        organizationNotAllowedOnProducerKeychain(
+          generateId(),
+          mockProducerKeychain.id
+        )
+      );
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token, mockProducerKeychain.id);
+    expect(res.status).toBe(403);
+  });
+
+  it("Should return 403 for organizationNotAllowedOnEService", async () => {
+    authorizationService.addProducerKeychainEService = vi
+      .fn()
+      .mockRejectedValue(
+        organizationNotAllowedOnEService(generateId(), mockEService.id)
+      );
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token, mockProducerKeychain.id);
+    expect(res.status).toBe(403);
+  });
```

Comment:
All of these tests could become a single it.each with an array of tuple error/expected status code